### ISSUE TITLE
no-unused-modules doesn't report unused Flow type exports

### ISSUE THREAD

**Issue by Hypnosphi** [benmosher/eslint-plugin-import#1540]
```js
// foo.js

// @flow strict
export type Foo = string
export default ''
```

```js
// index.js

// @flow strict
import foo from './foo'
```

Expected: `export type Foo` is reported as unused
Actual: no errors
